lib/cocoapods-search/command/search.rb

```diff
@@ -1,43 +1,97 @@
 module Pod
   class Command
-    # This is an example of a cocoapods plugin adding a top-level subcommand
-    # to the 'pod' command.
-    #
-    # You can also create subcommands of existing or new commands. Say you
-    # wanted to add a subcommand to `list` to show newly deprecated pods,
-    # (e.g. `pod list deprecated`), there are a few things that would need
-    # to change.
-    #
-    # - move this file to `lib/pod/command/list/deprecated.rb` and update
-    #   the class to exist in the the Pod::Command::List namespace
-    # - change this class to extend from `List` instead of `Command`. This
-    #   tells the plugin system that it is a subcommand of `list`.
-    # - edit `lib/cocoapods_plugins.rb` to require this file
-    #
-    # @todo Create a PR to add your plugin to CocoaPods/cocoapods.org
-    #       in the `plugins.json` file, once your plugin is released.
-    #
     class Search < Command
-      self.summary = 'Short description of cocoapods-search.'
+      self.summary = 'Search for pods.'
 
       self.description = <<-DESC
-        Longer description of cocoapods-search.
+        Searches for pods, ignoring case, whose name matches `QUERY`. If the
+        `--full` option is specified, this will also search in the summary and
+        description of the pods.
       DESC
 
-      self.arguments = 'NAME'
+      self.arguments = [
+        CLAide::Argument.new('QUERY', true),
+      ]
+
+      def self.options
+        [
+          ['--regex', 'Interpret the `QUERY` as a regular expression'],
+          ['--full',  'Search by name, summary, and description'],
+          ['--stats', 'Show additional stats (like GitHub watchers and forks)'],
+          ['--ios',   'Restricts the search to Pods supported on iOS'],
+          ['--osx',   'Restricts the search to Pods supported on OS X'],
+          ['--web',   'Searches on cocoapods.org'],
+        ].concat(super.reject { |option, _| option == '--silent' })
+      end 
 
       def initialize(argv)
-        @name = argv.shift_argument
+        @use_regex = argv.flag?('regex')
+        @full_text_search = argv.flag?('full')
+        @stats = argv.flag?('stats')
+        @supported_on_ios = argv.flag?('ios')
+        @supported_on_osx = argv.flag?('osx')
+        @web = argv.flag?('web')
+        @query = argv.arguments! unless argv.arguments.empty?
```

Comment:
The test "complains for wrong parameters" fails without the unless. 
See terminal output:

Pod::Command::Search
  Search
    ✓ registers it self (11 ms)
    ✓ runs with correct parameters (21 ms)
    - complains for wrong parameters [FAILED]
    ✓ searches for a pod with name matching the given query ignoring case (3 ms)